File: src/models/worker.go
package models

import (
	"time"
)

//go:generate reform
//reform:highkick_workers
type Worker struct {
	ID               int       `reform:"id,pk"`
	SID              string    `reform:"sid"`
	ProcessSID       string    `reform:"process_sid"`
	Stopped          bool      `reform:"stopped"`
	RunningJobsCount int       `reform:"running_jobs_count"`
	HealthcheckedAt  time.Time `reform:"healthchecked_at"`
	CreatedAt        time.Time `reform:"created_at"`
}

func (s *Worker) BeforeInsert() error {
	s.HealthcheckedAt = time.Now()
	s.CreatedAt = time.Now()
	return nil
}

File: client/src/models/worker.ts
import Moment from 'moment'

class Worker {
  ID!: number
  SID!: string
  ProcessSID!: string
  Stopped!: boolean
  RunningJobsCount!: number
  HealthcheckedAt!: string
  CreatedAt!: string

  constructor(props: Partial<Worker>) {
      for(const prop in props) {
          (this as any)[prop] = (props as any)[prop]
      }
  }

  isActive() {
    const lastHealthcheckAgo = Moment().diff(Moment.utc(this.HealthcheckedAt), 'seconds')
    return lastHealthcheckAgo < 90 // < 3 mins
  }
}

export default Worker